tests/aws/services/lambda_/test_lambda.py

```diff
@@ -1146,6 +1146,75 @@ def test_lambda_url_invocation_exception(self, create_lambda_function, snapshot,
         assert to_str(result.content) == "Internal Server Error"
         assert result.status_code == 502
 
+    @markers.aws.validated
+    def test_lambda_url_persists_after_alias_delete(
+        self, create_lambda_function, snapshot, aws_client
+    ):
+        snapshot.add_transformer(
+            snapshot.transform.key_value(
+                "FunctionUrl", "<function-url>", reference_replacement=False
+            )
+        )
+        function_name = f"test-fnurl-echo-{short_uid()}"
+        alias_name = f"alias-{short_uid()}"
+
+        create_lambda_function(
+            func_name=function_name,
+            handler_file=TEST_LAMBDA_URL,
+            runtime=Runtime.nodejs20_x,
+        )
+
+        fn_version_v1 = aws_client.lambda_.publish_version(FunctionName=function_name).get(
+            "Version"
+        )
+
+        aws_client.lambda_.create_alias(
+            FunctionName=function_name, FunctionVersion=fn_version_v1, Name=alias_name
+        )
+
+        aws_client.lambda_.get_waiter(waiter_name="function_updated_v2").wait(
+            FunctionName=function_name, Qualifier=alias_name
+        )
+
+        # Create an aliased URL
+        url_config = aws_client.lambda_.create_function_url_config(
+            FunctionName=function_name,
+            AuthType="NONE",
+            InvokeMode=InvokeMode.BUFFERED,
+            Qualifier=alias_name,
+        )
+        snapshot.match("create_lambda_url_config", url_config)
+
+        aws_client.lambda_.get_waiter(waiter_name="function_active_v2").wait(
+            FunctionName=function_name, Qualifier=alias_name
+        )
+
+        delete_alias_resp = aws_client.lambda_.delete_alias(
+            FunctionName=function_name, Name=alias_name
+        )
+        snapshot.match("delete_alias_resp", delete_alias_resp)
+
+        # Confirm the alias has deleted successfully
+        def _assert_alias_deleted():
+            assert (
+                len(
+                    aws_client.lambda_.list_aliases(
+                        FunctionName=function_name, FunctionVersion=fn_version_v1
+                    ).get("Aliases")
+                )
+                == 0
+            )
+
+        retry(_assert_alias_deleted, retries=15)
+
+        get_url_config = aws_client.lambda_.get_function_url_config(
+            FunctionName=function_name, Qualifier=alias_name
+        )
+        snapshot.match("get_url_config", get_url_config)
+
+        assert get_url_config.get("FunctionArn") == url_config.get("FunctionArn")
+        assert get_url_config.get("FunctionUrl") == url_config.get("FunctionUrl")
```

Comment:
What happens if we invoke the function URL after the alias is deleted?

Reply:
Just checked. We get a `403` forbidden. I believe the URL will be cleaned up asynchronously based on [the docs ](https://docs.aws.amazon.com/lambda/latest/dg/urls-configuration.html)(albeit the alias deletion is not the same as the function being deleted):

> When you delete a function that has a function URL, Lambda asynchronously deletes the function URL. If you immediately create a new function with the same name in the same account, it is possible that the original function URL will be mapped to the new function instead of deleted.

Reply:
Yeah 100%. Adding an invocation before and after the deletion 👍 

Perhaps it's not worth it but I'm thinking about how to test this asynchronous cleanup mechanism since the docs don't cover the above scenario. Maybe a sleep after delete, and then another test invoke?

Reply:
Yeah, for a quick validation, I would just add a long sleep and then set a breakpoint afterward so I could check interactively what an invoke returns. Once you know the actual behavior, it's easier to write a reliable test case using a retry until the expected condition is met.

Reply:
Just did a delayed test (waited 60 seconds after the alias was deleted) and it seems that the function URL gets cleaned up. Going to add a `retry` for this step.